### PR TITLE
Documentation for proposing a new market

### PR DESCRIPTION
**Pull request by teddyding**

### CODE REVIEW

File: pages/governance/proposing_a_new_market.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        }
+      },
+      {
+        "@type": "/dydxprotocol.delaymsg.MsgDelayMessage",
+        "authority": "dydx10d07y265gmmuvt4z0w9aw880jnsr700jnmapky",
+        "msg": {
+            "@type": "/dydxprotocol.clob.MsgUpdateClobPair",
+            "authority": "dydx1mkkvp26dngu6n8rmalaxyp3gwkjuzztq5zx6tr",
+            "clob_pair": {
+                "id": 0,
```
Comment (yang-dydx) [inline pages/governance/proposing_a_new_market.md:57]: They'd need to set this to the right id number right? How should we think about this (same with all the other `id`)

File: pages/governance/proposing_a_new_market.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                "perpetual_clob_metadata": {
+                  "perpetual_id": 0
+                },
+                "quantum_conversion_exponent": -9,
+                "status": "STATUS_ACTIVE",
+                "step_base_quantums": 1000000,
+                "subticks_per_tick": 100000
+            }
+        },
+        "delay_blocks" : 3600
```
Comment (yang-dydx) [inline pages/governance/proposing_a_new_market.md:67]: Is this the recommended value based on testing it on testnet? I assume we have some delay to account for all the daemons starting up etc. Research hasn't done much thinking on this

File: pages/governance/proposing_a_new_market.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
+# Proposing a new market
+
+## Example Proposal Json
+Below is an example proposal JSON file to propose adding `BTC-USD` as a new perpetual market (if it had not been added yet).
+
+```
+{
+    "title": "Add BTC-USD perpetual market",
+    "deposit": "10000000000000000000000adv4tnt",
```
Comment (yang-dydx) [inline pages/governance/proposing_a_new_market.md:9]: is this fixed or a recommendation?

File: pages/governance/proposing_a_new_market.md (hunk 1 above)
Comment (teddyding) [reply pages/governance/proposing_a_new_market.md]: Added some notes and updated example to `1001`. I explained that the `id` does not need to be consecutive as long as they are consistent for a market

File: pages/governance/proposing_a_new_market.md (hunk 2 above)
Comment (teddyding) [reply pages/governance/proposing_a_new_market.md:90]: Daemon doesn't need much bootstrap time (within a few seconds, perhaps) so the value here is really up to research recommendation

File: pages/governance/proposing_a_new_market.md (hunk 3 above)
Comment (teddyding) [reply pages/governance/proposing_a_new_market.md:32]: The required deposit, so I think the explanation is out-of-scope for this documentation. As an initial version for this doc I think it's fair to have sponsors figure this out (it's pretty straightforward for anyone who has made a proposal)

File: pages/governance/proposing_a_new_market.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+Notes:
+
+- For guideline on choosing values for individual params, see next [section](./proposing_a_new_market.md#choosing-market-parameters).
+- The exact ordering of messages above is necessary for successful onchain execution.
+- Each of the 4 top-level messages should have `Authority = dydx10d07y265gmmuvt4z0w9aw880jnsr700jnmapky`, the [gov module](https://github.com/dydxprotocol/v4-chain/blob/5e72896719e2f8d2fe6e10fddbde18b363a6bbe3/protocol/app/module_accounts_test.go#L28).
+- The `MsgCreateClobPair` message wrapped in `messages[3]: MsgDelayMessage` should have `Authority = dydx1mkkvp26dngu6n8rmalaxyp3gwkjuzztq5zx6tr`, the [delaymsg module](https://github.com/dydxprotocol/v4-chain/blob/5e72896719e2f8d2fe6e10fddbde18b363a6bbe3/protocol/app/module_accounts_test.go#L36).
+- The identifier fields must be consistent for a perpetual market: `params.id`, `params.market_id`, `clob_pair.id`, `clob_pair.perpetual_clob_metadata.perpertual_id`.  An `id` value is valid as long as it's a `uint32` unique from existing markets (they do not need to follow existing market ids).
+- `MsgDelayMessage.delay_blocks` is recommended to be `3600` blocks (`1 hour` at `1 sec` blocktime).
```
Comment (teddyding) [inline pages/governance/proposing_a_new_market.md:23]: Might sense to move this to the table below for a centralized view.

File: pages/governance/proposing_a_new_market.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+- Liquidity Tier (these are based on the liquidity tiers recommended, so mappings could change if these tiers change)
+  - 0: Large Cap (BTC/ETH)
+  - 1: Mid-Cap
+  - 2: Long-tail
+
+### Outputs
+
+| Message Type | Field | Description | Value |
+|--------------|-------|-------------|-------|
+| `MsgCreateOracleMarket` | `exponent` | TODO: definition | `p-9` |
```
Comment (teddyding) [inline pages/governance/proposing_a_new_market.md:48]: Definition was missing from research doc

File: pages/governance/proposing_a_new_market.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+- Safe (Not susceptible to market manipulation e.g. [Mango](https://www.notion.so/601722959e6d42ad8636f0e827885e9f?pvs=21))
+- Worth listing (Will this generate sufficient trading volume to consume throughput?)
+- Oracle sources are known (Which spot exchanges should be selected when determining the oracle price?)
+
+### Inputs
+
+- Ticker Symbol (e.g. `BTC-USD` )
+- Reference price (e.g. `40,000`) and `p` value
+  - Defines the exponent on the reference price as `log_price:=FLOOR(log10(reference_price))`. For example `log_price:=FLOOR(log10(40,000))=4`
+- Exchanges required for oracles (`exchange_config_json`)
```
Comment (teddyding) [inline pages/governance/proposing_a_new_market.md:38]: Should there be documentation on how this is set? IMO this is the most error prone part of the proposal

File: pages/governance/proposing_a_new_market.md (hunk 5 above)
Comment (yang-dydx) [reply pages/governance/proposing_a_new_market.md:48]: Definition: `Exponent denotes the number of decimals a value should be shifted in the price daemon.`

File: pages/governance/proposing_a_new_market.md (hunk 6 above)
Comment (yang-dydx) [reply pages/governance/proposing_a_new_market.md:38]: Agree this is most error prone, but also very nuanced so we are intentionally leaving it out of scope for this initial version.

File: pages/governance/proposing_a_new_market.md (hunk 4 above)
Comment (yang-dydx) [reply pages/governance/proposing_a_new_market.md:23]: Agree we should move this to the table

File: pages/governance/proposing_a_new_market.md (hunk 3 above)
Comment (yang-dydx) [reply pages/governance/proposing_a_new_market.md:64]: Makes sense

File: pages/governance/proposing_a_new_market.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+- The `MsgCreateClobPair` message wrapped in `messages[3]: MsgDelayMessage` should have `Authority = dydx1mkkvp26dngu6n8rmalaxyp3gwkjuzztq5zx6tr`, the [delaymsg module](https://github.com/dydxprotocol/v4-chain/blob/5e72896719e2f8d2fe6e10fddbde18b363a6bbe3/protocol/app/module_accounts_test.go#L36).
+- The identifier fields must be consistent for a perpetual market: `params.id`, `params.market_id`, `clob_pair.id`, `clob_pair.perpetual_clob_metadata.perpertual_id`.  An `id` value is valid as long as it's a `uint32` unique from existing markets (they do not need to follow existing market ids).
+- `MsgDelayMessage.delay_blocks` is recommended to be `3600` blocks (`1 hour` at `1 sec` blocktime).
+
+## Choosing Values for Market Parameters
+
+The following decribes how to set various parameters for a new market and assumes that the market listed is:
+
+- Safe (Not susceptible to market manipulation e.g. [Mango](https://www.notion.so/601722959e6d42ad8636f0e827885e9f?pvs=21))
```
Comment (yang-dydx) [inline pages/governance/proposing_a_new_market.md:29]: Realized for some reason i had an internal notion link. Could you update it to this: https://www.coindesk.com/tech/2022/10/20/defi-exchange-mangos-114m-exploit-was-market-manipulation-not-a-hack-ex-fbi-special-agent-says/

File: pages/governance/proposing_a_new_market.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+{
+    "title": "Add BTC-USD perpetual market",
+    "deposit": "10000000000000000000000adv4tnt",
+    "summary": "Add the `x/prices`, `x/perpetuals` and `x/clob` parameters needed for a BTC-UTC perpetual market. Create the market in `INITIALIZING` status and transition it to `ACTIVE` status after 3600 blocks."
+    "messages": [
+      {
+        "@type": "/dydxprotocol.prices.MsgCreateOracleMarket",
+        "authority": "dydx10d07y265gmmuvt4z0w9aw880jnsr700jnmapky",
+        "params": {
+            "exchange_config_json": '{"exchanges":[{"exchangeName":"Binance","ticker":"BTCUSDT","adjustByMarket":"USDT-USD"},{"exchangeName":"Bybit","ticker":"BTCUSDT","adjustByMarket":"USDT-USD"},{"exchangeName":"CoinbasePro","ticker":"BTC-USD"},{"exchangeName":"Huobi","ticker":"btcusdt","adjustByMarket":"USDT-USD"},{"exchangeName":"Kraken","ticker":"XXBTZUSD"},{"exchangeName":"Kucoin","ticker":"BTC-USDT","adjustByMarket":"USDT-USD"},{"exchangeName":"Mexc","ticker":"BTC_USDT","adjustByMarket":"USDT-USD"},{"exchangeName":"Okx","ticker":"BTC-USDT","adjustByMarket":"USDT-USD"}]}',
```
Comment (yang-dydx) [inline pages/governance/proposing_a_new_market.md:71]: Is something wrong with the JSON? Why are these lines highlighted? 
Screenshot 2023-12-13 at 3 51 09 PM

**Review by yang-dydx — APPROVED** [review]
LGTM, other than the comments I left. Thanks for copying it in.

File: pages/governance/proposing_a_new_market.md (hunk 3 above)
Comment (coreydydx) [reply pages/governance/proposing_a_new_market.md:64]: Should we reference the calculator anywhere that Yujin put together a while back? Think that could be helpful (if its still accurate) https://docs.google.com/spreadsheets/d/1DDOBjg3UPehSKiaw9g4l4NKNEuKd26Os6LtRe6gKlE0/edit#gid=0

File: pages/governance/proposing_a_new_market.md (hunk 6 above)
Comment (teddyding) [reply pages/governance/proposing_a_new_market.md:38]: Ok, I added a clear TODO item, which says more documentation is coming.

Comment (teddyding) [reply pages/governance/proposing_a_new_market.md:38]: Ok I added this as a line item in the table with a TODO, for visibility

File: pages/governance/proposing_a_new_market.md (hunk 3 above)
Comment (teddyding) [reply pages/governance/proposing_a_new_market.md:64]: > Should we reference the calculator anywhere that Yujin put together a while back?

We discussed this [here](https://www.notion.so/Proposing-a-new-market-4042cb25cef34707baf7058f19830f77?d=bf30c27299bd4c1a858c769bbc4df7fc&pvs=4#a9e8827897804222b6a8e91618de6c91).

File: pages/governance/proposing_a_new_market.md [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+
+### Inputs
+
+- Ticker Symbol (e.g. `BTC-USD` )
+- Reference price (e.g. `40,000`) and `p` value
+  - Defines the exponent on the reference price as `log_price:=FLOOR(log10(reference_price))`. For example `log_price:=FLOOR(log10(40,000))=4`
+- Exchanges required for oracles (`exchange_config_json`)
+- Liquidity Tier (these are based on the liquidity tiers recommended, so mappings could change if these tiers change)
+  - 0: Large Cap (BTC/ETH)
+  - 1: Mid-Cap
```
Comment (coreydydx) [inline pages/governance/proposing_a_new_market.md:40]: Should we define what we think mid-cap means? How does one determine mid cap vs small cap?

File: pages/governance/proposing_a_new_market.md (hunk 3 above)
Comment (coreydydx) [reply pages/governance/proposing_a_new_market.md:118]: ok, I personally think it's helpful to publish it (if its accurate). I find this file a bit confusing to figure out what the formulas are to get to the right outputs for some of these params

File: pages/governance/proposing_a_new_market.md (hunk 8 above)
Comment (teddyding) [reply pages/governance/proposing_a_new_market.md]: Fixed

**Pull request merged**